source/common/config/grpc_stream.h

```diff
@@ -78,6 +78,10 @@ class GrpcStream : public Grpc::TypedAsyncStreamCallbacks<ResponseProto>,
   void onReceiveMessage(std::unique_ptr<ResponseProto>&& message) override {
     // Reset here so that it starts with fresh backoff interval on next disconnect.
     backoff_strategy_->reset();
+    // Some times during hot restarts this stat's value becomes inconsistent and will continue to
+    // have 0 till it is reconnected. Setting here ensures that it is consistent with the state of
+    // management server connection.
+    control_plane_stats_.connected_state_.set(1);
```

Comment:
Where was it set previously? Should we remove it from there? Also is it possible to somehow test this change? NBD if not.

Reply:
It is set in `establishNewStream`. I think testing it may be difficult.

Reply:
Sorry. I missed your other question - I think  it is OK to leave it there because if there are no messages exchanged /delay in exchanges after stream establishment (which does not happen today) we will still have technically correct state of this stat that it is connected.

Reply:
OK sounds good.